backend/src/friend/friend.service.ts

```diff
@@ -1,4 +1,452 @@
+    }
+
+    // check if sender and receiver's sentRequest/receivedRequest array
+    // contain the corresponding email
+    if (
+      !sender.friendRequestsSent.includes(receiverEmail) ||
+      !receiver.friendRequestsReceived.includes(senderEmail)
+    ) {
+      throw new BadRequestException('Friend request not existant.');
+    }
+
+    // check if they are already friends
+    if (
+      sender.friends.includes(receiverEmail) ||
+      receiver.friends.includes(senderEmail)
+    ) {
+      throw new BadRequestException('You are already friends.');
+    }
+
+    // remove request from sender/receiver list
+    await this.removeRequestFromBothList(sender, receiver);
+
+    return {
+      status: 'OK',
+      statusCode: 200,
+      message: 'Friend request successufly canceled.',
+    };
+  }
+
+  /*****************************************************************************/
+  /* accept friend Request */
+  /*****************************************************************************/
+  async acceptFriendRequest(
+    user: User,
+    payload: { userEmail: string },
+    res: Response,
+  ) {
+    // check user/payload data
+    const checkData = await this.checkSenderAndReceiver(user, payload);
+    if (checkData !== '') {
+      throw new BadRequestException(checkData);
+    }
+
+    const senderEmail: string = payload.userEmail;
+    const receiverEmail: string = user.email;
+
+    const receiver: User = user;
+    const sender: User = await this.getUserByEmail(senderEmail);
+    if (!sender) {
+      throw new BadRequestException('Sender not found.');
+    }
+
+    // check if sender and receiver's sentRequest/receivedRequest array
+    // contain the corresponding email
+    if (
+      !sender.friendRequestsSent.includes(receiverEmail) ||
+      !receiver.friendRequestsReceived.includes(senderEmail)
+    ) {
+      throw new BadRequestException('Friend request not existant.');
+    }
+
+    // check if receiver was the sender
+    if (
+      receiver.friendRequestsSent.includes(senderEmail) ||
+      sender.friendRequestsReceived.includes(receiverEmail)
+    ) {
+      throw new BadRequestException('The receiver sent you an request.');
+    }
+
+    // check if they are already friends
+    if (
+      sender.friends.includes(receiverEmail) ||
+      receiver.friends.includes(senderEmail)
+    ) {
+      throw new BadRequestException('You are already friends.');
+    }
+
+    // remove request from sender/receiver list
+    await this.removeRequestFromBothList(sender, receiver);
+
+    // update/add sender to receiver's friends list
+    try {
+      const addFriendResult = await this.prisma.user.update({
+        where: {
+          email: receiverEmail,
+        },
+        data: {
+          friends: {
+            push: senderEmail,
+          },
+        },
+      });
+      if (addFriendResult.friends.length === 1) {
+        console.log('Added my first friend.');
+        // Add FRIEND achievement.
```

Comment:
Code to be added:
`user.achievements.push('FRIEND');`

I don't think a module is necessary, we don't neer more code than this if we avoid doing notifications.